lib/fluent/plugin/in_prometheus/async_wrapper.rb

```diff
@@ -13,8 +13,9 @@ def do_request(host:, port:, secure:)
             Async::HTTP::Endpoint.parse("http://#{host}:#{port}")
           end
 
-        client = Async::HTTP::Client.new(endpoint)
-        yield(AsyncHttpWrapper.new(client))
+        Async::HTTP::Client.open(endpoint) do |client|
+          yield(AsyncHttpWrapper.new(client))
+        end
```

Comment:
This ensures that any connected (persistent) connections are closed at the end of this block.

If you want to reuse this, you should write something like

```
# initialize
@client = Client.new(endpoint)

# each time:
yield @client
```

For long running process, you may want to have some upper limit on pool resources. Right now, this feature is not exposed in a simple way, but we plan on introducing automatic release of old/unused connections after a time.